### ISSUE TITLE
Size mismatch when loading pretrained models

### ISSUE THREAD

**Issue by benearnthof** [MICV-yonsei/EAGLE#6]
Hello and thank you for this great repository. 

I downloaded the checkpoints for cityscapes and cocostuff from the provided links but when I try to load the checkpoints like so: 

```python
model = LitUnsupervisedSegmenter.load_from_checkpoint(model_path, n_classes=27)
```

This leads to a size mismatch in the CELoss.learned_centroids layer of the model. In particular it seems to be initialized in the model (correctly) as torch.Size([27, 512]) but the checkpoint for cityscapes has a shape of torch.Size([28, 512]) for this layer, despite being initialized with 27 classes. This can be verified by checking:  

```python
chckpoint = torch.load("path_to_cityscapes.ckpt")
checkpoint["hyper_parameters"]["n_classes"]
```

This also yields 27 so I'm not quite sure what leads to this shape mismatch, as of right now I cannot evaluate the checkpoint provided by the authors.

**Comment by kochanha** [comment]
Hi, thank you for your interest in our work.
You can debug by changing code in modules.py line 464.

`self.learned_centroids = nn.Parameter(torch.randn(num_classes+1, cfg.dim))`

learned_centroids are not used for training so it won't make any difference to the final results.

**Comment by kochanha** [comment]
We've updated the code to the main repo. 
If you have any further questions, please open an issue or mail us. 
Thank you.

**Comment by benearnthof** [comment]
Thank you for addressing this so quickly I've pulled the recent changes and was able to verify the performance with the evaluation script. The metrics match the ViT_base/8 backbone numbers in the paper: 

```python
{
'final/linear/mIoU': 33.39485824108124, 
'final/linear/Accuracy': 91.37531518936157, 
'assignments': [...], 
'final/cluster/mIoU': 22.053803503513336, 
'final/cluster/Accuracy': 79.38949465751648
}
```
